### PR TITLE
discord bot service for arbitrary javascript evaluation

### PR DESCRIPTION
**Pull request by sbenthall**

### CODE REVIEW

**Comment by sbenthall** [comment]
This PR builds on #26, which should be merged first.

File: README.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 1. Move `.env-sample` to `.env` and replace `TOKEN` (ask @benheller)
 1. `nvm install` <-- select node version
 1. `yarn install` <-- install deps
+1.  E.g. `sudo service mongod start` if running mongo locally
```
Comment (benheller) [inline README.md:9]: If these changes aren't actually part of this PR (but rather #26), then please rebase this PR on top that branch so that the diff reflects only the changes introduced in this particular PR. This eliminates potential confusing around receiving feedback on the same code in multiple places.

File: app/services/discord/eval.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import Discord from 'discord.js';
+
+export default async function d_eval(message, opt) {
+
+    const out = eval(opt)
```
Comment (benheller) [inline app/services/discord/eval.js:5]: `eval` in Javascript is typically considered to be a major security issue, as it allows someone from Discord to run arbitrary code on the server in question. Since this server environment runs many other apps (some with sensitive information), this could be disastrous if merged. Note the banner here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval. As we say in the JS community "ev_al_ is ev_il_".

**Review by benheller — CHANGES_REQUESTED** [review]

**Comment by benheller** [comment]
I'm closing this pull request, as the intended functionality has the potential to be used maliciously.

**Comment by sbenthall** [comment]
I WAS TRYING TO EXPRESS MYSELF BEN

**Comment by sbenthall** [comment]
I know what eval is.
The question is: WHY would a user use it MALICIOUSLY?

File: README.md (hunk 1 above)
Comment (sbenthall) [reply README.md:9]: this is a pain in the bum

**Comment by sbenthall** [comment]
I apologize for my abrupt rudeness in this PR's comments. I should explain myself.

I am well aware of the reputation of `eval()` and I thought making this PR would be funny.

I think the functionality suggested (allowing the user to run javascript via the discord bot) is not on the whole bad. It would be a way to introduce non-coders to javascript. It looks like the linked documentation for `eval()`, there are some suggestions for how to do this in a safer and faster way. I think it would be worth thinking about them.

This is also an opportunity to disclose more about the production environment that PGBG is running in and to talk about how it may be made more secure. For example, _if_ it were running in a container, I believe it would be _more_ secure, and then it wouldn't matter so much that `eval()` was being used here.

If it's really a dealbreaking to update the live instance with the `eval()` code, I will change it. Of course I'm grateful for your review and don't want to waste your time. But I'd prefer to change it to something that didn't lose _all_ of its functionality.

In other words: I made this PR in the spirit of 'uncovery', as a move in PGBG/Event 5. I thought your response was a bit dogmatic! I merged the commit to continue the troll/game. I hope you will keep playing. It could get interesting.

**Comment by benheller** [comment]
Discussion continued at https://github.com/yrevocnu/psychedelic-glass-bead-game/issues/30#issuecomment-719932614
